kroxylicious-api/src/main/java/io/kroxylicious/proxy/plugin/PluginConfigType.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.plugin;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that identifies the config type consumed by a specific plugin implementation.
+ * This is applied to the class implementing a plugin interface.
+ * Use @PluginConfigType(Void.class) if a plugin implementation class doesn't require configuration.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface PluginConfigType {
```

Comment:
Why is this a separate annotation to `PluginConfig` Isn't this a 2nd property of `PluginCongfig`

Reply:
The `@PluginConfig` is used at the plugin point (e.g. the `FilterDefinition`), where nothing it known about the class used to represent the config for any particular plugin implementation. 

This annotation is used on a plugin implementation class to say what type of config type it consumes. It's basically a replacement for the configType() method. It was necessary to change that because we need to tell Jackson what the type is, prior to having an instance of the plugin implementtion (knowing only its class).